scripts/character_test.js

```diff
@@ -1,26 +1,20 @@
-/*global $ validCharacterPattern XRegExp */
-/* exported testText */
+/*global XRegExp goodChar */
+/* exported makeGoodCharRegex testText */
 
 // regex unicode property escape is supported in Chrome and Safari (Feb 2020)
 // but not in Firefox or Edge. Use 3rd party http://xregexp.com/ instead
 
 // this matches any character: non-mark codepoint followed by 0 or more marks
 const charMatch = XRegExp("\\PM\\pM*", "Ag");
 
-// this regular expression (constructed below) matches individual good characters
-var goodChar;
-
-$(function () {
-    // need to define this after the page has loaded so validCharacterPattern
-    // is available
-    goodChar = XRegExp(validCharacterPattern, "A");
-});
+function makeGoodCharRegex(validCharacterPattern) {
```

Comment:
`makeValidCharRegex()` would be a better name.

Reply:
Done this.